orca/2.1-a-first-look-at-a-neural-network.ipynb

```diff
@@ -0,0 +1,518 @@
+    "\n",
+    "Then we need to define two configuration parameters, which are:\n",
+    "\n",
+    "* `max_epoch`: the maximum number of passes of the entire training dataset.\n",
+    "*  `batch_size`: the number of training examples utilized in one iteration.\n",
+    "\n",
+    "As an example, we can define them as follows:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "max_epoch = 5\n",
+    "batch_size = 320"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "\n",
+    "Before training, we will preprocess our data by reshaping it into the shape that the network expects, and scaling it so that all values are in \n",
+    "the `[0, 1]` interval. We need to create a trainging and a testing dataset whose elements are slices of the original datasets.\n",
+    "Also, our training images for instance were stored in an array of shape `(60000, 28, 28)` of type `uint8` with \n",
+    "values in the `[0, 255]` interval previously. We need to transform it into a `float32` array of shape `(60000, 28 * 28)` with values between 0 and 1. In addition, the \"repeat\" and \"shuffle\" process will help the model avoid bias and converge faster."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "def preprocess(x, y):\n",
+    "    x = tf.cast(tf.reshape(x, (28, 28, 1)), dtype=tf.float32) / 255.0\n",
+    "    return x, y\n",
+    "\n",
+    "def train_data_creator(config):\n",
+    "    (train_feature, train_label), _ = mnist.load_data()\n",
+    "\n",
+    "    dataset = tf.data.Dataset.from_tensor_slices((train_feature, train_label))\n",
+    "    dataset = dataset.repeat()\n",
+    "    dataset = dataset.map(preprocess)\n",
+    "    dataset = dataset.shuffle(1000)\n",
+    "    dataset = dataset.batch(config[\"batch_size\"])\n",
+    "    return dataset\n",
+    "\n",
+    "def val_data_creator(config):\n",
+    "    _, (val_feature, val_label) = mnist.load_data()\n",
+    "\n",
+    "    dataset = tf.data.Dataset.from_tensor_slices((val_feature, val_label))\n",
+    "    dataset = dataset.repeat()\n",
+    "    dataset = dataset.map(preprocess)\n",
+    "    dataset = dataset.batch(config[\"batch_size\"])\n",
+    "    return dataset"
+   ]
+  },
+  {
+   "source": [
+    "We are now ready to train our network using Orca Estimator. Firstly, we need to initialize the Orca context by specifying local or distributed mode."
+   ],
+   "cell_type": "markdown",
+   "metadata": {}
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [
+    
+   ],
+   "source": [
+    "from zoo.orca import init_orca_context, stop_orca_context\n",
+    "from zoo.orca import OrcaContext\n",
+    "\n",
+    "OrcaContext.log_output = False  # (True: display terminal's stdout and stderr in the Jupyter notebook).\n",
```

Comment:
The default value of `log_output` is false, so maybe you do not need to set it.

Reply:
Ok, I can remove this line.

Reply:
Just set it to true (with comments for explanation)

Reply:
Ok.